TrainingCenter/models/Course.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace TrainingCenter.models;
+
+public class Course
+{
+    private string title;
+    private List<Lesson> lessons;
+    private List<Teacher> teachers;
+
+    public Course(string title)
+    {
+        this.title = title;
+        this.lessons = new List<Lesson>();
+        this.teachers = new List<Teacher>();
+    }
+
+    public Course(string title, List<Lesson> lessons)
+    {
+        this.title = title;
+        this.lessons = lessons;
```

Comment:
Broken encapsulation, outside the class will be the reference to this collection. Here you should create copy of this list for example using list constructor